packages/block-library/src/query/view.js

```diff
@@ -18,83 +24,70 @@ const isValidEvent = ( event ) =>
-					) {
-						event.preventDefault();
+			if ( isValidLink( ref ) && isValidEvent( event ) && ! isDisabled ) {
+				event.preventDefault();
 
-						const id = ref.closest( '[data-wp-navigation-id]' )
-							.dataset.wpNavigationId;
+				const id = ref.closest( '[data-wp-navigation-id]' ).dataset
+					.wpNavigationId;
 
-						// Don't announce the navigation immediately, wait 400 ms.
-						const timeout = setTimeout( () => {
-							context.core.query.message =
-								context.core.query.loadingText;
-							context.core.query.animation = 'start';
-						}, 400 );
+				// Don't announce the navigation immediately, wait 400 ms.
+				const timeout = setTimeout( () => {
+					ctx.message = ctx.loadingText;
+					ctx.animation = 'start';
+				}, 400 );
 
-						await navigate( ref.href );
+				yield navigate( ref.href );
 
-						// Dismiss loading message if it hasn't been added yet.
-						clearTimeout( timeout );
+				// Dismiss loading message if it hasn't been added yet.
+				clearTimeout( timeout );
 
-						// Announce that the page has been loaded. If the message is the
-						// same, we use a no-break space similar to the @wordpress/a11y
-						// package: https://github.com/WordPress/gutenberg/blob/c395242b8e6ee20f8b06c199e4fc2920d7018af1/packages/a11y/src/filter-message.js#L20-L26
-						context.core.query.message =
-							context.core.query.loadedText +
-							( context.core.query.message ===
-							context.core.query.loadedText
-								? '\u00A0'
-								: '' );
+				// Announce that the page has been loaded. If the message is the
+				// same, we use a no-break space similar to the @wordpress/a11y
+				// package: https://github.com/WordPress/gutenberg/blob/c395242b8e6ee20f8b06c199e4fc2920d7018af1/packages/a11y/src/filter-message.js#L20-L26
+				ctx.message =
+					ctx.loadedText +
+					( ctx.message === ctx.loadedText ? '\u00A0' : '' );
 
-						context.core.query.animation = 'finish';
-						context.core.query.url = ref.href;
+				ctx.animation = 'finish';
+				ctx.url = ref.href;
 
-						// Focus the first anchor of the Query block.
-						const firstAnchor = `[data-wp-navigation-id=${ id }] .wp-block-post-template a[href]`;
-						document.querySelector( firstAnchor )?.focus();
-					}
-				},
-				prefetch: async ( { ref } ) => {
-					const isDisabled = ref.closest( '[data-wp-navigation-id]' )
-						?.dataset.wpNavigationDisabled;
-					if ( isValidLink( ref ) && ! isDisabled ) {
-						await prefetch( ref.href );
-					}
-				},
-			},
+				// Focus the first anchor of the Query block.
+				const firstAnchor = `[data-wp-navigation-id=${ id }] .wp-block-post-template a[href]`;
+				document.querySelector( firstAnchor )?.focus();
+			}
+		},
+		*prefetch() {
+			const { ref } = getElement();
+			const isDisabled = ref.closest( '[data-wp-navigation-id]' )?.dataset
+				.wpNavigationDisabled;
+			if ( isValidLink( ref ) && ! isDisabled ) {
+				yield prefetch( ref.href );
+			}
 		},
 	},
 	effects: {
```

Comment:
Rename to `callbacks`.

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/55459/commits/3010ab12a1c17242f3e8a1e8a5e0ee403c6a42e4.